M/MAGMA/build_tarballs.jl

```diff
@@ -1,38 +1,60 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
 name = "MAGMA"
-version = v"2.6.2"
+version = v"2.7.0"
+
+# Collection of sources required to complete build
 sources = [
-    GitSource("https://bitbucket.org/icl/magma", "864fa8b28e8c2905a41fbef32f156695378dc104"),
-    DirectorySource("./bundled")
+    ArchiveSource("http://icl.utk.edu/projectsfiles/magma/downloads/magma-2.7.0.tar.gz", "fda1cbc4607e77cacd8feb1c0f633c5826ba200a018f647f1c5436975b39fd18"),
+    DirectorySource("./bundled"),
+    # 10.x isn't supported apparently.
+    # ArchiveSource("https://github.com/JuliaBinaryWrappers/CUDA_full_jll.jl/releases/download/CUDA_full-v10.2.89%2B5/CUDA_full.v10.2.89.x86_64-linux-gnu.tar.gz", "60e6f614db3b66d955b7e6aa02406765e874ff475c69e2b4a04eb95ba65e4f3b"; unpack_target = "CUDA_full.v10.2"),
+    ArchiveSource("https://github.com/JuliaBinaryWrappers/CUDA_full_jll.jl/releases/download/CUDA_full-v11.3.1%2B1/CUDA_full.v11.3.1.x86_64-linux-gnu.tar.gz", "9ae00d36d39b04e8e99ace63641254c93a931dcf4ac24c8eddcdfd4625ab57d6"; unpack_target = "CUDA_full.v11.3")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/magma/
-
+cd $WORKSPACE/srcdir/magma-*
+cuda_version=`echo $bb_full_target | sed -E -e 's/.*cuda\+([0-9]+\.[0-9]+).*/\1/'`
+cuda_version_major=`echo $cuda_version | cut -d . -f 1`
+cuda_version_minor=`echo $cuda_version | cut -d . -f 2`
+cuda_full_path="$WORKSPACE/srcdir/CUDA_full.v$cuda_version/cuda"
+export PATH=$PATH:${cuda_full_path}/bin
+export CUDADIR=${cuda_full_path}
 cp ../make.inc .
-
-make -j${nproc} sparse-lib
+(( nproc=1+nproc/3 ))
```

Comment:
As always, it'd be useful to leave comments to explain decisions which would otherwise look cumbersome, especially if the decision (for the record: reduce parallelism to try and make successful the compilation of large code which would otherwise run out of memory) was taken offline